### PR TITLE
SuccessAnimation: use state reducer pattern

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Try out the state reducer pattern for `SuccessAnimation` component.

## Justify why they are needed

I think it clarifies things and cleanly separates logic from presentation.

Read more here: https://kentcdodds.com/blog/the-state-reducer-pattern-with-react-hooks

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1849** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1849" target="_blank">Graphite</a> 
    * **PR #1854** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1854" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1854?utm_source=stack-comment).

File: apps/store/src/components/ConfirmationPage/SuccessAnimation.tsx [hunk 1]
```diff
@@ -65,3 +58,48 @@ const VideoWrapper = styled(motion.div)({
...
+    case ActionType.ANIMATE:
+      return 'animation'
+
+    case ActionType.COMPLETE:
+      return 'complete'
+
+    default:
+      return state
+  }
+}
```
Comment (KarnellSchultz) [inline apps/store/src/components/ConfirmationPage/SuccessAnimation.tsx:82]: Fun stuff right here

**Review by KarnellSchultz — APPROVED** [review]
rad-movie-gif

File: apps/store/src/components/ConfirmationPage/SuccessAnimation.tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/components/ConfirmationPage/SuccessAnimation.tsx:82]: haha you like or no like?

**Comment by robinandeer** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Job ID: UVEMsRZiRPr0aGB3WPGQ.
See details on [graphite.dev](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1854)

**Pull request merged**